twake/backend/node/src/services/documents/services/index.ts

```diff
@@ -653,22 +653,37 @@ export class DocumentsService {
       zlib: { level: 9 },
     });
 
+    let counter = ids.length;
+
     await Promise.all(
       ids.map(async id => {
         if (!(await checkAccess(id, null, "read", this.repository, context))) {
           this.logger.warn(`not enough permissions to download ${id}, skipping`);
+          counter--;
           return;
         }
 
         try {
-          await addDriveItemToArchive(id, null, archive, this.repository, context);
+          counter = await addDriveItemToArchive(
+            id,
+            null,
+            archive,
+            this.repository,
+            context,
+            undefined,
+            counter,
```

Comment:
```suggestion
            counter,
```

```suggestion
            undefined,
```

probably we can swap the prefix and counter params order since counter is always passed and the prefix is optional.
something like:
```
export const addDriveItemToArchive = async (
  id: string,
  entity: DriveFile | null,
  archive: archiver.Archiver,
  repository: Repository<DriveFile>,
  context: CompanyExecutionContext,
  counter: number,
  prefix?: string,
): Promise<number> => {
```